infra/examples-dev/aws-all/variables.tf

```
@@ -53,6 +53,12 @@ variable "aws_ssm_param_root_path" {
   }
 }
 
+variable "secrets_store_implementation" {
+  type        = string
+  description = "one of 'aws_ssm_parameter_store' (default) or 'aws_secrets_manager'"
+  default     = "aws_ssm_parameter_store"
```

Adding validation with a constraint for avoiding random values here? (Same for other places where this new parameter is used)

```suggestion
  default     = "aws_ssm_parameter_store"
  
  validation {
    condition     = contains(["aws_ssm_parameter_store", "aws_secrets_manager"], var.secrets_store_implementation)
    error_message = "Allowed values for secrets_store_implementation are \"aws_ssm_parameter_store\" or \"aws_secrets_manager\"."
  }
```